packages/number-field/test/number-field.test.js

```diff
@@ -0,0 +1,884 @@
+      it('should set value to min when min is above zero and max is above zero and increaseButton is clicked', () => {
+        numberField.min = 3;
+        numberField.max = 19;
+        numberField.step = 6;
+
+        increaseButton.click();
+
+        expect(numberField.value).to.be.equal('3');
+      });
+
+      it('should set value to min when min is above zero and max is below zero and increaseButton is clicked', () => {
+        numberField.min = 19;
+        numberField.max = -3;
+        numberField.step = 6;
+
+        increaseButton.click();
+
+        expect(numberField.value).to.be.equal('19');
+      });
+
+      it(`should set value to the first step value above zero when min is below zero
+          and max is above zero and increaseButton is clicked`, () => {
+        numberField.min = -19;
+        numberField.max = 19;
+        numberField.step = 6;
+
+        increaseButton.click();
+
+        expect(numberField.value).to.be.equal('5');
+      });
+
+      it('should set value to max when min is below zero and max is below zero and decreaseButton is clicked', () => {
+        numberField.min = -19;
+        numberField.max = -3;
+        numberField.step = 6;
+
+        decreaseButton.click();
+
+        expect(numberField.value).to.be.equal('-3');
+      });
+
+      it('should set value to min when min is above zero and max is above zero and decreaseButton is clicked', () => {
+        numberField.min = 3;
+        numberField.max = 19;
+        numberField.step = 6;
+
+        decreaseButton.click();
+
+        expect(numberField.value).to.be.equal('3');
+      });
+
+      it('should set value to max when min is above zero and max is below zero and decreaseButton is clicked', () => {
+        numberField.min = 19;
+        numberField.max = -3;
+        numberField.step = 6;
+
+        decreaseButton.click();
+
+        expect(numberField.value).to.be.equal('-3');
+      });
+
+      it(`should set value to the first step value below zero when min is below zero and max
+          is above zero and decreaseButton is clicked`, () => {
+        numberField.min = -19;
+        numberField.max = 19;
+        numberField.step = 6;
+
+        decreaseButton.click();
+
+        expect(numberField.value).to.be.equal('-1');
+      });
+    });
+
+    describe('min and max values are undefined', () => {
+      it('should set value to the first step value above zero when decreaseButton is clicked', () => {
+        numberField.step = 6;
+
+        increaseButton.click();
+
+        expect(numberField.value).to.be.equal('6');
+      });
+
+      it('should set value to the first step value below zero when decreaseButton is clicked', () => {
+        numberField.step = 6;
+
+        decreaseButton.click();
+
+        expect(numberField.value).to.be.equal('-6');
+      });
+    });
+  });
+
+  describe('input validation', () => {
+    it('should be valid with numeric values', () => {
+      expect(numberField.validate()).to.be.true;
```

Comment:
~~**nit:** Would suggest extracting this assertion into a separate test with a name like:~~

~~`should be valid with initial properties` or `should be valid by default`~~

**Update:** This part is just a copy-paste from the old version of the NumberField.